### ISSUE TITLE
'Table index is out of bounds' and 'Memory access out of bounds' on Android Chrome

### ISSUE THREAD

**Issue by Ben-Mack** [google/draco#726]
Recently, Chrome on Android starts reporting some strange errors when decoding Draco:
`Memory access out of bounds`
or
`table index is out of bounds`
Resulted in models can't be loaded into scene. The models is just a few dozens tris so it's probably not caused by memory limit.

It happens most of the time, try reloading sometime works, sometime it won't, and it affects a large percentage of our models (that never experience these errors before).
Reproduced on Note FE, S8+, S20+ with latest Chrome.

On Android, it only occurs with Chrome (currently v91). Firefox and Samsung Internet are all fine.
iOS and Windows also no problem.

If I uninstall Chrome, it will reverted back to the older version installed by default (v74), and that version doesn't having these errors at all.

I'm using Draco 1.4.1, tried to revert back to 1.3.6 but the problems still persisted.

**Comment by Ben-Mack** [comment]
This is a reproducable sandbox: https://codesandbox.io/s/draco-android-chrome-bug-forked-uiskd?file=/src/index.js

If you don't have the devices, you can use [BrowserStack open source](https://www.browserstack.com/open-source), select Galaxy S10 Chrome and visit https://uiskd.csb.app/ to validate the problem

The sandbox contains both the original uncompressed glTF and Draco compressed glTF.
You can see the error only occurs on Draco version.

**Comment by FrankGalligan** [comment]
Hi @Ben-Mack,

This is actually an issue with WASM in M91 of Chrome Android.

You can read more about the issue and Draco [here](https://github.com/google/model-viewer/issues/2496).
You can read more about the underlying issue [here](https://bugs.chromium.org/p/v8/issues/detail?id=11863).

Unfortunately as of today the issue is not fixed, but the does not exist in M92, which is shipping to stable tomorrow (I think).

We put up a version 1.4.2 on gstatic which can workaround the issue for some models, but not all. Here is the info for changing model-viewer to use 1.4.2:
https://www.gstatic.com/draco/versioned/decoders/1.4.2/

Basically this is 1.4.1 but we set INITIAL_MEMORY to 64mb. So this is only a workaround, because if a specific models needs > 64MB memory to decode then we will hit the issue again.

Can you try setting model-viewer to point to 1.4.2? E.g.
ModelViewerElement.dracoDecoderLocation = 'https://www.gstatic.com/draco/versioned/decoders/1.4.2/';

And it's a static property, so don't forget this line: ModelViewerElement = self.ModelViewerElement || {}; https://modelviewer.dev/examples/loading/#dracoSupport

**Comment by FrankGalligan** [comment]
I'm closing this issue as other issues like this were closed with Chrome Android M92.

**Comment by Ben-Mack** [comment]
Draco 1.4.2 has this error in console when load high poly mesh (~2 mil tris)
```
{ 
type: "error", 
id: 1, 
error: "abort(undefined). Build with -s ASSERTIONS=1 for more info." 
}
```
Using Note FE and Xiaomi K20 pro, reproduced on both Chrome 92 and Chrome Dev 95. (Free Ram is more than 1gb)
Firefox and Samsung Internet works fine.

Can you upload the build with `-s ASSERTIONS=1` so I can test if it gives more info? (tried building by myself on Visual Studio 2019 but not successful)

**Comment by Ben-Mack** [comment]
This is a reproduceable case:
https://codesandbox.io/s/draco-android-chrome-bug-forked-2ydjz

Here I load 2 gltf file with 2.5 million tris each.
(In my app, only one 2.5 mil tris glTF is enough to make this abort error occurred. However, to reproduce on this CodeSandBox, it need 2 file to make the error occurred)

**Comment by Ben-Mack** [comment]
@FrankGalligan I tried to use JS version as workaround, then I realized JS decoder also sufferred by similar error (even in Chrome 95) 
Codesandbox for JS version: https://codesandbox.io/s/draco-android-chrome-bug-forked-xi96s

Could you please check if there're any fix for this `abort(undefined)` error?
![image](https://user-images.githubusercontent.com/6309393/133352690-ff751b7d-52df-474e-bd34-e15266237346.png)

**Comment by Ben-Mack** [comment]
@FrankGalligan Could you please reopen this issue because I'm still having this problem with the latest Draco 1.5.0 and Chrome 97 on different devices S21, S8, Note FE, with both Draco WASM and JS decoder.
